extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -210,4 +211,18 @@ public void handle(RoutingContext event) {
             }
         }
     }
+
+    private static boolean isSameOrigin(HttpServerRequest request, String origin) {
+        String absUriString = request.absoluteURI();
+        if (absUriString.startsWith(origin)) {
```

Comment:
Are we sure `origin` ends with a `/` and we wouldn't have a problem with `http://my-origin` and `http://my-origin-wrong-domain`?

Reply:
@gsmet This is just a first check - then we make sure this Origin has the same scheme, host and port as the current request URI

Reply:
Note that this will fail if the origin has the port and the host does not.